### PR TITLE
fix(security): redact token-shaped values from sandbox doctor --json output

### PR DESCRIPTION
**Pull request by Dongni-Yang**
## Summary

`sandbox doctor --json` resolved with the raw report, while `sandbox status --json` has redacted its machine-readable report since #4310. Both doctor JSON egress points now pass through the centralized `redactForLog`, so token-shaped values in check details no longer reach JSON consumers. Exit codes are still computed from the raw report before redaction.

## Related Issue

Refs #4310 (the `status --json` redaction precedent this change mirrors). The asymmetry was found while verifying the redaction contract for the serving-process health design discussion on #7003 (refs only; this PR does not close either issue).

## Changes

- `src/lib/actions/sandbox/doctor-report.ts`: the `renderDoctorReport` JSON branch prints via `console.log`, which bypasses the oclif `logJson` redaction boundary (#3657); the report is now routed through `redactForLog` before printing.
- `src/commands/sandbox/doctor.ts`: the oclif `--json` path now returns `redactForLog(report)`, so programmatic consumers of the resolved value — not just the `logJson`-printed stdout — never see raw token-shaped detail. `process.exitCode` is derived from the raw report first.
- Tests: one render-level and one command-boundary redaction test. Each uniquely pins its egress point: the command test mocks `runSandboxDoctor` (so `renderDoctorReport` never runs on that path), and the render test calls `renderDoctorReport` directly (the oclif path bypasses it via `quietJson`). Both were red before the fix.

Review notes:

- Pre-existing quirk (unchanged here, identical in `status --json`): `redactFull`'s word-context patterns can consume a short word following a credential-shaped env-var name, e.g. `…_API_KEY for NVIDIA managed health…` renders as `…_API_KEY <REDACTED> NVIDIA managed health…`. A candidate follow-up is to give `FULL_REDACT_PATTERNS` the same `{10,}` minimum value length `CONTEXT_PATTERNS` already uses.
- The `renderDoctorReport` JSON branch currently has no production caller (the oclif command always passes `quietJson: true` with `--json`); it is redacted anyway because that `console.log` egress has no other redaction layer for any future direct caller.

## Type of Change

- [x] Code change (feature, bug fix, or refactor)
- [ ] Code change with doc updates
- [ ] Doc only (prose changes, no code sample modifications)
- [ ] Doc only (includes code sample changes)

## Quality Gates

- [x] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [ ] Tests not applicable — justification:
- [ ] Docs updated for user-facing behavior changes
- [x] Docs not applicable — justification: `doctor --json` is documented only as "Emit the report as JSON" with no field-level or verbatim-detail contract (`docs/reference/commands.mdx`); this change enforces the existing CLI redaction contract in `docs/security/best-practices.mdx`.
- [x] Sensitive paths changed (security, policy, credentials, preflight, onboarding, inference, runner, sandbox, or messaging)
- [x] Sensitive-path review completed or maintainer-approved waiver recorded — reviewer/approval link/justification: [cjagwani approved PR SHA `92aa402`](https://github.com/NVIDIA/NemoClaw/pull/7356#pullrequestreview-4756008102) after reviewing the redaction coverage, resolved automated feedback, required checks, and selected E2E.
- [ ] Non-success, skipped, or missing CI check accepted by maintainer — check name, approval link, and follow-up issue:

## DGX Station Hardware Evidence

- [ ] Tested on DGX Station
- Tested commit: Not applicable
- Station profile/scenario: Not applicable
- Result: Not applicable
- Supporting evidence: Not applicable — this PR does not change `scripts/prepare-dgx-station-host.sh`.

## Verification

- [x] PR description includes a `Signed-off-by:` line and every commit appears as `Verified` in GitHub
- [x] Normal `pre-commit`, `commit-msg`, and `pre-push` hooks passed, or `npm run check:diff` passed when hooks were skipped or unavailable — `npm run check:diff` passed (hooks skipped locally; commitlint reports one non-blocking `footer-leading-blank` warning)
- [x] Targeted behavior tests pass for the current change set, or tests are marked not applicable above — `npx vitest run --project cli src/lib/actions/sandbox/doctor-report.test.ts src/commands/sandbox/oclif-command-adapters.test.ts src/lib/actions/sandbox/doctor-flow.test.ts src/lib/actions/sandbox/doctor-inference.test.ts` → 42/42 passed
- [ ] Applicable broad gate passed — `npm test` for broad runtime/test-harness changes; `npm run check` for repo-wide validation/coverage changes — command/result: not applicable (targeted change)
- [x] Quality Gates section completed with required justifications or waivers
- [x] No secrets, API keys, or credentials committed

---
Signed-off-by: Dongni-Yang <dongniy@nvidia.com>

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**

  * Sandbox diagnostic reports in JSON mode now redact token-shaped values from returned and displayed details.
  * Preserved failure status reporting while preventing sensitive token values from appearing in machine-readable output.

* **Tests**

  * Added coverage confirming redaction in both CLI results and printed JSON reports.

### CODE REVIEW

File: src/lib/actions/sandbox/doctor-report.test.ts [hunk 1]
```diff
@@ -34,6 +34,27 @@ describe("doctor reports", () => {
...
+    ]);
+
+    expect(renderDoctorReport(report, true)).toBe(1);
+    const printed = JSON.parse(String(logSpy.mock.calls[0]?.[0]));
+    expect(printed.checks[0].detail).toBe(
+      "connect failed: Authorization: Bearer <REDACTED> (HTTP 401)",
+    );
+    expect(printed.checks[0].hint).toBe("restart the gateway");
+    expect(JSON.stringify(printed)).not.toContain("sk-abc123DEF456ghi789");
+  });
```
[the remark this answers is not in the document]
Comment (Dongni-Yang) [reply src/lib/actions/sandbox/doctor-report.test.ts:56]: The spy is already restored: this `describe` block registers `afterEach(() => vi.restoreAllMocks())` (line 12, unchanged in this PR), which runs even when an assertion in the test throws — so the `console.log` spy cannot leak into later tests. The new test intentionally follows the same pattern as the sibling test above it ("renders the machine-readable report and returns a failing exit code", lines 29–35), which spies on `console.log` the same way and relies on the same hook. Adding a per-test `try/finally` + `mockRestore()` here would duplicate the hook and diverge from the file's established style, so leaving as-is.

**Review by cjagwani — APPROVED** [review]
Reviewed 92aa402697eb1362ea8597dfc7b6e8ddde084f89. The token redaction fix is covered, the only CodeRabbit thread is resolved/withdrawn, both review-advisor lanes report no actionable findings, and all required CI plus selected E2E passed. Approving; not merging.
